### ISSUE TITLE
Allow Asset Manager API to return size

### ISSUE THREAD

**Issue by chrislo** [alphagov/asset-manager#482]
The `AssetManagerAndQuarentinedFileStorage` engine implements a `size` method that used the quarentined file to calculate the size. 

We plan to switch the storage engine for AttachmentUploader to `AssetManagerStorage` and therefore store all files exclusively in Asset Manger. This means we will need to implment the `size` method on `AssetManagerStorage` and it will have to fetch the data from the Asset Manager API. on remove all of the assets from the disk, so we need to instead calculate it by making a call to the Asset Manager API. We have [migrated all the existing attachments to Asset Manager](https://github.com/alphagov/asset-manager/issues/441) and are adding new ones to both the disk and AM. That means we can make this change now - it will make it easier to subsequently [store the attachments exlusively in Asset Manager](https://github.com/alphagov/asset-manager/issues/470). 

- [x] Store file size information in the Asset Manager Database when a new asset is added
- [x] Add file size information to the JSON returned by Asset Manager
- [x] Run `db:set_size_for_all_uploaded_assets` on `integration`
- [x] Run `db:set_size_for_all_uploaded_assets` on `staging`
- [x] Run `db:set_size_for_all_uploaded_assets` on `production`

**Comment by chrislo** [comment]
The code in PR #483 has been deployed to production, so we are now saving asset sizes for newly uploaded assets.

**Comment by chrislo** [comment]
The worker-based migration added in #484 is currently been run on integration and staging.

**Comment by chrislo** [comment]
The migration took just under an hour to run (rake task + processing all jobs) on `staging`. It's considerably slower on integration (around 150 job/s compared to almost 3000 job/s on staging). 

<img width="1408" alt="screen shot 2018-02-20 at 10 22 21" src="https://user-images.githubusercontent.com/16707/36432353-00e49f44-1628-11e8-907e-12024bfcfddb.png">

I've asked 2ndline to run the task on `production`.

**Comment by chrislo** [comment]
The migration completed in around 80 mins on production. 

<img width="1409" alt="screen shot 2018-02-20 at 11 34 52" src="https://user-images.githubusercontent.com/16707/36436417-283ae17a-1632-11e8-8b80-130d3edf9b68.png">

**Comment by chrislo** [comment]
And, it completed in around 4.5 hours on integration. 

<img width="1411" alt="screen shot 2018-02-20 at 12 52 36" src="https://user-images.githubusercontent.com/16707/36440145-0093a548-163d-11e8-94da-379ceb2c9c63.png">

**Comment by chrislo** [comment]
Following continued work switching Whitehall AttachmentUploader to use the AssetManagerStorage engine, which prompted me to create this issue, I've discovered that we don't yet need to fetch the size of the asset into Whitehall. It was used to validate replaced assets, but I've found a way to avoid having to fetch the actual size. The size of the attachment is also displayed to the public, but Whitehall currently caches that in its database on initial upload. 

I think it would be good to use size from Asset Manager rather than a cached version in Whitehall as it would mean we could make attachments behave more like other asset types. So for now, I don't think its worthwhile removing the size field I added in this issue from Asset Manager, but I am going to close this issue.
